Quorum/apis/price_feeds/price_feed_utils.py

```diff
@@ -40,32 +60,36 @@ def __init__(self):
         self.cache_dir = Path(__file__).parent / "cache" / self.get_name().value
         self.cache_dir.mkdir(parents=True, exist_ok=True)
         self.session = requests.Session()
-        self.memory: dict[Chain, dict[str, PriceFeedData]] = {}
+        self.memory: dict[str, PriceFeedData] = {}
 
-    def get_feeds(self, chain: Chain) -> dict[str, PriceFeedData]:
+    def get_price_feed(self, chain: Chain, address: str) -> PriceFeedData | None: 
         """
-        Get price feed data for a given blockchain network from the cache.
+        Get price feed data for a given address on a blockchain network.
 
         Args:
             chain (Chain): The blockchain network to fetch price feed data for.
+            address (str): The contract address of the price feed.
 
         Returns:
-            dict[str, PriceFeedData]: A dictionary mapping the contract address of the price feed to the price feed data.
+            PriceFeedData: The price feed data for the specified address or None if not found.
         """
-        cache_file = self.cache_dir / f"{chain.value}.json"
+        cache_file = self.cache_dir / f"{chain.value}" / f"{address}.json"
         if cache_file.exists():
             with open(cache_file, 'r') as file:
                 data: dict = json.load(file)
-            self.memory[chain] = {addr: PriceFeedData(**feed) for addr, feed in data.items()}
+            self.memory[address] = PriceFeedData(**data)
         else:
-            if chain not in self.memory:
-                self.memory[chain] = self._get_price_feeds_info(chain)
+            if address not in self.memory:
+                self.memory[address] = self._get_price_feed_info(chain, address)
+                if not self.memory[address]:
+                    return None
+                cache_file.parent.mkdir(parents=True, exist_ok=True)
                 with open(cache_file, 'w') as file:
-                    json.dump({addr: feed.dict() for addr, feed in self.memory[chain].items()}, file)
-        return self.memory[chain]
+                    json.dump(self.memory[address].model_dump(mode="json"), file)
```

Comment:
```suggestion
                    json.dump(self.memory[address].model_dump(mode="json"), file, indent=4)
```
For readability if we'll ever need it.